R/PipeOpFDATSFeatures.R

```diff
@@ -0,0 +1,108 @@
+#' * `trim` :: `logical(1)`\cr
+#' If `TRUE`, data is trimmed by `trim_amount` before features are computed.
+#' Values larger than `trim_amount` in absolute value are set to `NA`. Default is `FALSE`.
+#' * `trim_amount` :: `numeric(1)`\cr
+#' Default level of trimming. Default is `0.1`.
+#' * `parallel` :: `logical(1)`\cr
+#' If `TRUE`, the features are computed in parallel. Default is `FALSE`.
+#' * `multiprocess` :: `any`\cr
+#' The function from the future package to use for parallel processing. Default is `future::multisession`.
+#' * `na.action` :: `any`\cr
+#' A function to handle missing values. Default is `stats::na.pass`.
+#'
+#' @section Naming:
+#' The new names generally append a `_{feature}` to the corresponding column name.
+#' If a column was called `"x"` and the feature is `"trend"`, the corresponding new column will
+#' be called `"x_trend"`.
+#'
+#' @export
+#' @examples
+#' task = tsk("fuel")
+#' po_tsfeats = po("fda.tsfeats")
+#' task_tsfeats = po_tsfeats$train(list(task))[[1L]]
+#' task_tsfeats$data()
+PipeOpTSFeatures = R6Class("PipeOpTSFeatures",
+  inherit = PipeOpTaskPreprocSimple,
+  public = list(
+    #' @description Initializes a new instance of this Class.
+    #' @param id (`character(1)`)\cr
+    #'   Identifier of resulting object, default is `"fda.tsfeats"`.
+    #' @param param_vals (named `list`)\cr
+    #'   List of hyperparameter settings, overwriting the hyperparameter settings that would
+    #'   otherwise be set during construction. Default `list()`.
+    initialize = function(id = "fda.tsfeats", param_vals = list()) {
+      param_set = ps(
+        features = p_uty(
+          default = c("frequency", "stl_features", "entropy", "acf_features"),
+          tags = "train",
+          custom_check = crate(function(x) check_character(x, any.missing = FALSE, min.len = 1L))
+        ),
+        scale = p_lgl(default = TRUE, tags = c("train", "predict")),
+        trim = p_lgl(default = FALSE, tags = c("train", "predict")),
+        trim_amount = p_dbl(
+          default = 0.1, tags = c("train", "predict"), depends = quote(trim == TRUE)
+        ),
+        parallel = p_lgl(default = FALSE, tags = c("train", "predict")),
+        multiprocess = p_uty(
+          default = future::multisession,
+          tags = c("train", "predict"),
+          depends = quote(parallel == TRUE),
+          custom_check = check_function
+        ),
+        na.action = p_uty(
+          default = stats::na.pass,
+          tags = c("train", "predict"),
+          custom_check = check_function
+        )
+      )
+
+      super$initialize(
+        id = id,
+        param_set = param_set,
+        param_vals = param_vals,
+        packages = c("mlr3fda", "mlr3pipelines", "tf", "tsfeatures"),
+        feature_types = c("tfd_reg", "tfd_irreg"),
+        tags = "fda"
+      )
+    }
+  ),
+
+  private = list(
+    .transform_dt = function(dt, levels) {
+      require_namespaces(
```

Comment:
This should not be necessary when `"tsfeatures"` is added to `packages`

Reply:
But I've put it under suggests, then I would have to direct put it in imports

Reply:
I don't fully understand your point

Reply:
Simply that I don't want to put it in imports, but then its already being checked if its installed.